tools/runners/run-app.js

```diff
@@ -404,9 +404,15 @@ Object.assign(AppRunner.prototype, {
     self.startPromise = self._makePromise("start");
 
     self.isRunning = true;
-    self._runApp().catch((e) => self._resolvePromise("start", e));
-    await self.startPromise;
-    self.startPromise = null;
+    try {
+      await self._runApp();
```

Comment:
I'm not sure about this one. I did several tests around this code and I'm pretty sure the version you changed is the final version.

In some cases is ok to wait for  `self._runApp()` to finish, but there are other cases where the process can hang there forever. 

I'm not sure why you had to do this change, but we can look into it later. Pretty sure the problem comes from another place.

Reply:
Found the real problem, we were not cleaning the context for this function when it was running. Before, we were running it inside a new `Fiber` which would have a clean context. The fix here is to use the `asyncLocalStorage.run({}, fn)` to reset the context for this function.